mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/search/DeletedMessageField.java

```diff
@@ -41,13 +44,89 @@
 
     }
 
-    DeletedMessageField<ZonedDateTime> DELETION_DATE = () -> ValueExtractor.DELETION_DATE_EXTRACTOR;
-    DeletedMessageField<ZonedDateTime> DELIVERY_DATE = () -> ValueExtractor.DELIVERY_DATE_EXTRACTOR;
-    DeletedMessageField<Collection<MailAddress>> RECIPIENTS = () -> ValueExtractor.RECIPIENTS_EXTRACTOR;
-    DeletedMessageField<MailAddress> SENDER = () -> ValueExtractor.SENDER_EXTRACTOR;
-    DeletedMessageField<Boolean> HAS_ATTACHMENT = () -> ValueExtractor.HAS_ATTACHMENT_EXTRACTOR;
-    DeletedMessageField<Collection<MailboxId>> ORIGIN_MAILBOXES = () -> ValueExtractor.ORIGIN_MAILBOXES_EXTRACTOR;
-    DeletedMessageField<String> SUBJECT = () -> ValueExtractor.SUBJECT_EXTRACTOR;
+    DeletedMessageField<ZonedDateTime> DELETION_DATE = new DeletedMessageField<ZonedDateTime>() {
```

Comment:
I don't see the point of having different types here anymore, it makes everything very verbose when building instances with parameters would make it more readable